pkg/query-service/app/http_handler.go

```diff
@@ -321,7 +321,8 @@ func (aH *APIHandler) RegisterRoutes(router *mux.Router) {
 	router.HandleFunc("/api/v1/services", ViewAccess(aH.getServices)).Methods(http.MethodPost)
 	router.HandleFunc("/api/v1/services/list", aH.getServicesList).Methods(http.MethodGet)
 	router.HandleFunc("/api/v1/service/overview", ViewAccess(aH.getServiceOverview)).Methods(http.MethodPost)
-	router.HandleFunc("/api/v1/service/top_endpoints", ViewAccess(aH.getTopEndpoints)).Methods(http.MethodPost)
+	router.HandleFunc("/api/v1/service/top_operations", ViewAccess(aH.getTopOperations)).Methods(http.MethodPost)
+	router.HandleFunc("/api/v1/service/top_level_operations", ViewAccess(aH.getServicesTopLevelOps)).Methods(http.MethodPost)
```

Comment:
It's generally recommended to use use `-` instead of `_` in api's .
https://metamug.com/article/rest-api-naming-best-practices.html#5-hyphen-and-resource-nesting-as-delimiters
https://developers.google.com/search/docs/advanced/guidelines/url-structure

Reply:
I think links talk about the URL an user sees. I think `_` is fine and we have been using it many places. Initially to provide the same endpoint as prom https://prometheus.io/docs/prometheus/latest/querying/api/#range-queries.

Reply:
@srikanthccv We have been using camelCase for API endpoints in most places in signoz so I would suggest to use camelCase to maintain consistency.

Reply:
I have seen it's usually `_` separated in the path unless it's targeted for webcrawler. Orgs like [Stripe](https://stripe.com/docs/api) and [Reddit](https://www.reddit.com/dev/api/) known for their best developer facing APIs also follow that. Sharing couple of links who also does that. I don't have strong opinion but I think we should prefer using `_`.
https://tinyurl.com/2p9d9jy5
https://tinyurl.com/2bc6eum6

What do you think?

Reply:
Feel free to follow new practices since we should continuously evolve to using better practices rather than sticking to leagacy codebase practices

Reply:
Sure, @srikanthccv let's document this on notion as best practices which have to be followed.

Reply:
Created an issue [here](https://github.com/SigNoz/engineering-pod/issues/631) to track and address it separately.